added an error-handling middleware

niyobern opened the pull request:
**What does this PR do?**
This PR adds a global error-handling middleware that will handle all uncaught errors in controllers and api routes

**Description of Task to be completed?**
defined error-handling middleware

**How should this be manually tested?**

- import the middleware function into main file

`import errorHandler from '/middlewares/errorHandler'
`

- add the middleware as the last middleware in the application with: `app.use(errorhandler)`

**Run Commands**
npm run dev
**Any background context you want to provide?**
I am not logging down errors but sending errors to the client

ambroisegithub reviewed (commented):
That's is Look Good!!!

Habinezajanvier commented on src/index.js:
```diff
@@ -0,0 +1,18 @@
+"use strict";
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+const express_1 = __importDefault(require("express"));
+const cors_1 = __importDefault(require("cors"));
+const morgan_1 = __importDefault(require("morgan"));
+const app = (0, express_1.default)();
+app.use((0, cors_1.default)());
+app.use((0, morgan_1.default)("dev"));
+app.get("/", (req, res) => {
+    return res.json({ message: "Welcome To The Dynamites backend e-commerce" });
+});
+const PORT = 3000;
+app.listen(PORT, () => {
+    console.log(`The App is running on ${PORT}`);
+});
```
Why do we have this one here

niyobern replied:
It originated from origin/develop when I did git pull

niyobern commented:
I am not the one who wrote this file. I pulled it from develop

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: HABINEZA Janvier ***@***.***>
Sent: Monday, April 29, 2024 6:07:04 PM
To: atlp-rwanda/dynamites-ecomm-be ***@***.***>
Cc: NIYOMUGABO BERNARD ***@***.***>; Assign ***@***.***>
Subject: Re: [atlp-rwanda/dynamites-ecomm-be] added an error-handling middleware (PR #23)


@Habinezajanvier commented on this pull request.

________________________________

In src/index.js<https://github.com/atlp-rwanda/dynamites-ecomm-be/pull/23#discussion_r1583346196>:

> +"use strict";
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+const express_1 = __importDefault(require("express"));
+const cors_1 = __importDefault(require("cors"));
+const morgan_1 = __importDefault(require("morgan"));
+const app = (0, express_1.default)();
+app.use((0, cors_1.default)());
+app.use((0, morgan_1.default)("dev"));
+app.get("/", (req, res) => {
+    return res.json({ message: "Welcome To The Dynamites backend e-commerce" });
+});
+const PORT = 3000;
+app.listen(PORT, () => {
+    console.log(`The App is running on ${PORT}`);
+});


Why do we have this one here

—
Reply to this email directly, view it on GitHub<https://github.com/atlp-rwanda/dynamites-ecomm-be/pull/23#pullrequestreview-2028945926>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AUKJPRIW7M7G3FYPHW5GZIDY7ZV2RAVCNFSM6AAAAABGZENL3CVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMRYHE2DKOJSGY>.
You are receiving this because you were assigned.Message ID: ***@***.***>

wayneleon1 committed:
Set Up Testing Framework and Unit Tests (#22)

* init commit

Configure Jest for TypeScript & Add a test script to package.json

Create a tests directory for unit tests

rebasing from develop

Fix conflicts

rebasing from develop

changes before rebase

* commenting

* Refactor folder structure and update database connection

* create new test-db for testing mock data

* forcing output.log in gitignore

* setting up test
